### PR TITLE
Adding new PR naming standards and processes to contribution

### PR DESCRIPTION
**Pull request by Jerry-Jinfeng-Guo**
This PR adds the agreed content to the contribution guide.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -145,7 +145,14 @@ The process for a code change and pull request you should follow:
 "feature/###" or "fix/###". For more information see the Git branching guideline.
 1. Make changes, compile, and test thoroughly. Ensure any install or build dependencies are removed before the end of the layer when doing a build. Code style should match existing style and conventions, and changes should be focused on the topic the pull request will be addressed. For more information see the style guide.
 1. Push commits to your fork.
-1. Create a Github pull request from your topic branch.
+1. Create a Github pull request from your topic branch, following the steps below:
+    1. Name your PR with "{Feature or EPIC name}" + ": " + "{title for this particular PR}": e.g., "Step-up transformer tap changer support: updated ranking"
```
Comment (mgovers) [inline CONTRIBUTING.md:149]: what if there's no associated feature? e.g. https://github.com/PowerGridModel/power-grid-model/pull/825 (`Fix nightly build`). Would it be something along the lines of `Nightly build: fix`?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -145,7 +145,14 @@ The process for a code change and pull request you should follow:
...
+    2. Assign yourself to this PR
+    3. Optionally request reviews from reviewers
+    4. **Add label** to the PR
+    5. **Do not** specify Project: leave it blank
```
Comment (mgovers) [inline CONTRIBUTING.md:153]: please do mention elsewhere in this file that all issues shall be assigned to the PGM Project

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -145,7 +145,14 @@ The process for a code change and pull request you should follow:
...
+    6. If there is an issue / feature request, link them by either mentioning "Closes ###" or selecting it in Development
+        1. Check in the issue / feature request to see whether the link works
```
Comment (mgovers) [inline CONTRIBUTING.md:155]: please mention that if if doesn't actually **close** the issue, then mentioning it as e.g. `Relates to` is also fine (and cross-link from the issue to the PR as well, if applicable)

File: CONTRIBUTING.md (hunk 1 above)
Comment (mgovers) [inline CONTRIBUTING.md:149]:
```suggestion
    1. Name your PR with "<Feature or Epic>: <title for this PR>": e.g., "Step-up transformer tap changer support: updated ranking"
```

Comment (Jerry-Jinfeng-Guo) [reply CONTRIBUTING.md:149]: I think `{}` are more natural place holders

Comment (mgovers) [reply CONTRIBUTING.md:149]: in that case
```suggestion
    1. Name your PR with "{Feature or Epic}: {title for this PR}": e.g., "Step-up transformer tap changer support: updated ranking"
```

Comment (Jerry-Jinfeng-Guo) [reply CONTRIBUTING.md:149]: I really don't think a general guideline should include all edge cases but in this case I would say `Nightly build` fits in the category of feature or epic and should indeed go along those lines. I will mention that "{Feature or EPIC name}: " is not as strictly required when there is no bigger feature that the PR is belonging to.

**Review by TonyXiang8787 — APPROVED** [review]

**Pull request merged**